### PR TITLE
Graylog integration

### PR DESCRIPTION
**Pull request by Hruuttila**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Integration with Graylog product. (https://www.graylog.org/)

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Graylog/Integrations/Graylog/Graylog.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    else:
+        start_time = timefrom
+    end_time = datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3]
+    end_time += 'Z'
+    parameters = {'query': incidentquery,
+                  'from': start_time,
+                  'to': end_time}
+    results = client._http_request('GET', '/search/universal/absolute', params=parameters)
+    if 'total_results' in results and results['total_results'] > 0:
+        demisto.setLastRun({'start_time': datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3]})
```
Comment (orlichter1) [inline Packs/Graylog/Integrations/Graylog/Graylog.py:71]: In order not to lose data in the time it takes to query the results please use `end_time` as the new `start_time` you are setting.

**Review by orlichter1 — CHANGES_REQUESTED** [review]
Please change the integration logo to fit the [standard](https://xsoar.pan.dev/docs/integrations/integration-logo)

File: Packs/Graylog/Integrations/Graylog/Graylog.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        elif demisto.command() == 'graylog-events-search':
+            jsonparameterswithvalue = {}
+            jsonparameters = {'query': demisto.args().get('query'),
+                              'filter': demisto.args().get('filter'),
+                              'page': demisto.args().get('page'),
+                              'sort_direction': demisto.args().get('sort_direction'),
+                              'per_page': demisto.args().get('per_page'),
+                              'timerange': {'type': 'relative', 'range': demisto.args().get('timerange')},
+                              'sort_by': demisto.args().get('sort_by')}
+            # API does not like None as the key value so getting rid of those.
```
[the remark this answers is not in the document]
Comment (Hruuttila) [reply Packs/Graylog/Integrations/Graylog/Graylog.py:146]: Hi, Great! Did not know such function exists. Changed code to use it.

File: Packs/Graylog/Integrations/Graylog/Graylog.py (hunk 1 above)
Comment (Hruuttila) [reply Packs/Graylog/Integrations/Graylog/Graylog.py:71]: Yeah, good point, a bit of a thinking error on my part there. Changed start_time to be the end_time for the next round.

File: Packs/Graylog/Integrations/Graylog/README.md [hunk 3]
```diff
@@ -0,0 +1,1205 @@
...
+
+#### Context Output
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| Graylog.Search | String | Search results | 
+
+
+#### Command Example
+```!graylog-search query=<query here>```
```
Comment (orlichter1) [inline Packs/Graylog/Integrations/Graylog/README.md:352]: One of the errors raised in `ci/circleci: Run Validations` is that `<query here>` is perceived as a part of an HTML code that isn't closed. In order to remedy the error please replace `<query here>` with `\<query here\>` everywhere it appears.

**Review by orlichter1 — APPROVED** [review]
The validation step fails now as it's missing 2 files:
`Packs/Graylog/.secrets-ignore`
`Packs/Graylog/.pack-ignore`
Once you add them this step should pass and we can merge this contribution 😄

**Pull request merged**